pna_control/pna_control.py

```diff
@@ -116,7 +138,44 @@ def powersweep(startpower: float, endpower: float, numsweeps: int, centerf: floa
     file.write('POINTS: '+str(points)+'\n')
     file.close()
 
-    #run each sweep while increasing averages for each power
+    #run each sweep
     for i in sweeps:
-        getdata(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        get_data(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        # Iterate number of averages as power decreases
         averages = averages * ((10**(stepsize/10))**0.5)
+
+def name_datafile(outputfile: str,
+                  power: float,
+                  temp: float) -> str:
+    filename = outputfile+'_'+str(power)+'dB'+'_'+str(temp)+'mK'
+    filename = filename.replace('.','p')
+
+    return filename
+    
+def timestamp_folder(dir: str, meastype: str) -> str:
+    """Create a filename and directory structure to annotate the scan.
+
+        Takes a root directory, appends scan type and timestamp.
+
+        Args:
+            dir: root directory for the scan
+            meastype: type of measurements, eg: 'powersweep' 
+
+        Returns:
+            Formatted path eg. dir/5p51414GHz_HPsweep_200713_12_18_04/ 
+    """
+    now = time.strftime("%y%m%d_%H_%M_%S", time.localtime())
+
+    output = meastype+ '_' + now
+    output = output.replace('.','p')
+    
+    if dir != None:
```

Comment:
Should `dir` be a keyword argument with a default value of `None` when not specified by the user?
This would make sense if the user wanted to give a relative or absolute path?
Also, it might be a good idea to check that the path exists by adding something like `if len(glob.glob(dir))`.

Reply:
Good suggestions. I added the first. For the second, I think that's covered under line 178:
> while os.path.isdir(output_path):
Let me know if that's not the case.